core/src/test/java/org/elasticsearch/search/internal/SearchSortValuesTests.java

```diff
@@ -0,0 +1,139 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.internal;
+
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode;
+
+public class SearchSortValuesTests extends ESTestCase {
+
+    public static SearchSortValues createTestItem() {
+        List<Supplier<Object>> valueSuppliers = new ArrayList<>();
+        // this should reflect all values that are allowed to go through the transport layer
+        valueSuppliers.add(() -> null);
+        valueSuppliers.add(() -> randomInt());
+        valueSuppliers.add(() -> randomLong());
+        valueSuppliers.add(() -> randomDouble());
+        valueSuppliers.add(() -> randomFloat());
+        valueSuppliers.add(() -> randomByte());
+        valueSuppliers.add(() -> randomShort());
+        valueSuppliers.add(() -> randomBoolean());
+        valueSuppliers.add(() -> frequently() ? randomAsciiOfLengthBetween(1, 30) : randomRealisticUnicodeOfCodepointLength(30));
+
+        int size = randomInt(20);
+        Object[] values = new Object[size];
+        for (int i = 0; i < size; i++) {
+            Supplier<Object> supplier = randomFrom(valueSuppliers);
+            values[i] = supplier.get();
+        }
+        return new SearchSortValues(values);
+    }
+
+    public void testFromXContent() throws IOException {
+        SearchSortValues sortValues = createTestItem();
+        // exclude SMILE: its difficult to compare exact float values after parsing
+        // they get returned as doubles with a slightly different value if not cast back to float (which we cannot do due to
+        // losing type information on the rest layer)
```

Comment:
I changed this using the common helpers you pointed me to, left TODOs where we need to remove those once they are moved to a common class.